### PR TITLE
Add some per-query histograms to the chunk store.

### PR DESCRIPTION
**Pull request by tomwilkie**
Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

### CODE REVIEW

File: pkg/chunk/series_store.go [hunk 1]
```diff
@@ -112,12 +143,19 @@ func (c *seriesStore) lookupSeriesByMetricNameMatchers(ctx context.Context, from
...
+			preIntersectionPerQuery.Observe(float64(len(series)))
+			postIntersectionPerQuery.Observe(float64(len(series)))
+		}
+		return series, err
 	}
 
 	// Otherwise get series which include other matchers
 	incomingIDs := make(chan []string)
 	incomingErrors := make(chan error)
+	indexLookupsPerQuery.Observe(float64(len(matchers)))
```
Comment (gouthamve) [inline pkg/chunk/series_store.go:158]: `len(matchers) + 1` for the `metricName`?

File: pkg/ingester/ingester.go [hunk 2]
```diff
@@ -68,7 +68,7 @@ var (
 		Name: "cortex_ingester_queried_chunks",
 		Help: "The total number of chunks returned from queries.",
 		// A small number of chunks per series - 10*(4^8) = 655k.
-		Buckets: prometheus.DefBuckets,
+		Buckets: prometheus.ExponentialBuckets(10, 4, 8),
```
Comment (gouthamve) [inline pkg/ingester/ingester.go:71]: 655K is overkill no?

**Review by gouthamve — COMMENTED** [review]
LGTM except the two nits.

File: pkg/chunk/series_store.go (hunk 1 above)
Comment (tomwilkie) [reply pkg/chunk/series_store.go:158]: We don't do a lookup on metric name, its in the row key.

File: pkg/ingester/ingester.go (hunk 2 above)
Comment (tomwilkie) [reply pkg/ingester/ingester.go:71]: I'm thinking 100k series and 6 chunks per series which is only 18hrs...

File: pkg/chunk/series_store.go [hunk 3]
```diff
@@ -112,12 +143,19 @@ func (c *seriesStore) lookupSeriesByMetricNameMatchers(ctx context.Context, from
 
 	// Just get series for metric if there are no matchers
 	if len(matchers) == 0 {
-		return c.lookupSeriesByMetricNameMatcher(ctx, from, through, metricName, nil)
+		indexLookupsPerQuery.Observe(float64(1))
```
Comment (gouthamve) [inline pkg/chunk/series_store.go]: You don't need to cast to `float64`. And if there is no `+1` below, then this would cause `0` lookups right?

**Pull request merged**